src/core/src/service.rs

```diff
@@ -240,29 +241,83 @@ impl Poet2Service {
+                wait_certificate_sig, 
+                poet2_util::to_hex_string(summary),
+                wait_time
+            );
+
+        let mut payload_to_send = serial_cert;
+        payload_to_send.push_str("#");
+        payload_to_send.push_str(&cert_signature); 
+        return payload_to_send.clone();
+    }
+
+    pub fn verify_wait_certificate( &mut self, _block: &Block, previous_block: &Block, poet_pub_key: &String) -> bool {
+        let mut block = _block.clone();
+        let mut wait_cert_verify_status:bool = false;
+
+        
+        let (wait_cert, wait_cert_sign) = get_wait_cert_and_signature(&block);
+        debug!("Serialized wait_cert : {:?}", &wait_cert);
+        let deser_wait_cert:WaitCertificate = serde_json::from_str(&wait_cert).unwrap();
+
+        let sig_verify_status = EnclaveConfig::verify_wait_certificate(self.enclave.enclave_id,
+                                                            &poet_pub_key,
+                                                            &wait_cert, &wait_cert_sign);
+
+        debug!("sig_verify_status={:?}", sig_verify_status);
+
+        let prev_id = poet2_util::blockid_to_hex_string(block.previous_id);
+        let block_id = poet2_util::blockid_to_hex_string(block.block_id);
+        let signer_id =poet2_util::to_hex_string(block.signer_id.to_vec());
+        let summary = poet2_util::to_hex_string(block.summary);
+        
+        if( (deser_wait_cert.prev_block_id == prev_id) &&
+            (deser_wait_cert.block_number == block.block_num) &&
+            (deser_wait_cert.block_summary == summary) &&
+            (deser_wait_cert.validator_id == signer_id) &&
+            sig_verify_status) {
+            
+            if( block.block_num > 1){
+                let mut prev_block = previous_block.clone();
+                let (prev_wait_cert, prev_wait_cert_sig) = get_wait_cert_and_signature(&prev_block);
+                if(deser_wait_cert.prev_wait_cert_sig == prev_wait_cert_sig){
+                    wait_cert_verify_status = true;
+                }
+            }
+            else if(block.block_num == 1) {
+                wait_cert_verify_status = true;
+            }
         }
-         info!("Block id returned is {:?}", Vec::from(chain_head.block_id.clone()));
-         let (serial_cert, cert_signature) = EnclaveConfig::finalize_wait_certificate(
-                 self.enclave.enclave_id,
-                 self.enclave.signup_info,
-                 poet2_util::blockid_to_hex_string(chain_head.previous_id),
-                 prev_wait_certificate_sig, 
-                 poet2_util::to_hex_string(summary),
-                 wait_time
-             );
-
-         let mut payload_to_send = serial_cert;
-         payload_to_send.push_str("#");
-         payload_to_send.push_str(&cert_signature); 
-         return payload_to_send.clone();
+
+        debug!("wait_cert_verify_status = {:?}", wait_cert_verify_status);
```

Comment:
Enhance log message to something more readable - Certificate verification failed/Certificate verification succeeded

Reply:
ok